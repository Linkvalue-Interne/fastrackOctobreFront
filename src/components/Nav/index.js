// == Import : npm
import React from 'react';
import styled from '@emotion/styled';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faUser, faGraduationCap, faChartLine, faDoorOpen } from '@fortawesome/free-solid-svg-icons';

// Import local

import Navbar from '../commons/nav';

// CSS

const IconContainer = styled.div`width: 80px;
height: 100vh;
display: flex;
justify-content: space-around;
align-items: center;
flex-direction: column;`;

const Img = styled.img`width: 80px;
position: absolute;
top: 0;`;

// == Composant

const Nav = () => (
  <Navbar>
    <Img src="src/styles/img/Logo-LV.png" alt="" />
    <IconContainer>
      <Link><FontAwesomeIcon icon={faUser} style={{ color: '#FFF', width: '80px' }} size="2x" /></Link>
      <Link><FontAwesomeIcon icon={faGraduationCap} style={{ color: '#FFF', width: '80px' }} size="2x" /></Link>
      <Link><FontAwesomeIcon icon={faChartLine} style={{ color: '#FFF', width: '80px' }} size="2x" /></Link>
    </IconContainer>
    <Link><FontAwesomeIcon icon={faDoorOpen} style={{ color: '#FFF', width: '80px', position: 'absolute', top: 'calc(100% - 48px)' }} size="1x" /></Link>
  </Navbar>
);

// == Export

export default Nav;
