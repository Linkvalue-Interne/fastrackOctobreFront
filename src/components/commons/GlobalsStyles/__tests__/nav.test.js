import React from 'react';
import renderer from 'react-test-renderer';
import serializer from 'jest-emotion';

import Nav from '../nav';

expect.addSnapshotSerializer(serializer);

describe('Nav', () => {
  it('renders with correct style', () => {
    const tree = renderer.create(<Nav />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
