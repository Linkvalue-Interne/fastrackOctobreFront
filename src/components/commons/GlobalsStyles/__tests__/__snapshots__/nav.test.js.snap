// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders with correct style 1`] = `
.emotion-0 {
  width: 80px;
  height: 100vh;
  background: #29AAE2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 10rem 0;
  position: fixed;
}

@media screen and (max-width:549px) {
  .emotion-0 {
    position: absolute;
    width: 100%;
    height: 50px;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    padding: 0 0;
    top: 0;
  }
}

<div
  className="emotion-0"
/>
`;
